functional-programming/src/fp/utils.ts

```diff
@@ -0,0 +1,55 @@
+import { Maybe, none, some } from "./maybe";
+
+export const constant = <A>(a: A) => () => a;
+
+/**
+ * Performs left-to-right function composition
+ */
+export function flow<A, B, C>(fa: (a: A) => B, fb: (b: B) => C): (a: A) => C;
+export function flow<A, B, C, D>(fa: (a: A) => B, fb: (b: B) => C, fc: (c: C) => D): (a: A) => D;
+export function flow<A, B, C, D, E>(fa: (a: A) => B, fb: (b: B) => C, fc: (c: C) => D, fd: (d: D) => E): (a: A) => E;
+export function flow(...fns: Array<(...args: Array<string>) => string>) {
+  return (a: string) => fns.reduce(
+    (acc, fn) => fn(acc),
+    a
+  );
+}
```

Comment:
task from my side)

write `flow` function that can run async functions one by one

use case: u have array of urls stings and u need to execute fetching of data one by one and accumulate data from each.